### PR TITLE
Add a setting to disable save document

### PR DESCRIPTION
**Pull request by Hzj-jie**
This is a small change to add a hidden setting to disable save document feature.

I am considering whether we should add an individual 'control panel' like module.

### CODE REVIEW

File: frontend/apps/reader/readerui.lua [hunk 1]
```diff
@@ -465,18 +465,23 @@ end
...
-            cancel_text = _("No"),
-            ok_callback = function()
-                self:closeDocument()
-            end,
-            cancel_callback = function()
-                self.document:discardChange()
-                self:closeDocument()
-            end,
-        })
+        if G_reader_settings:readSetting("disable_save_document") then
```
Comment (chrox) [inline frontend/apps/reader/readerui.lua]: Is there any reason why this option is needed considering that user has an opportunity to discard change in the dialog?

Comment (Hzj-jie) [inline frontend/apps/reader/readerui.lua]: From my own opinion, this function is not really useful. Currently only PdfDocument implemented this function, but it writes highlight to the original file. Considering highlight is also stored in metadata file, and as a reader, I believe in most cases, we discard changes.

**Comment by jedick** [comment]
For me this is a welcome change. I am a habitual highlighter, but I never want the highlights saved to the PDF, so I find the "Do you want to save" dialog to be annoying.

Conversely, I always want to export all new or altered highlights to HTML (via the Evernote plugin). It would be nice to do that without having to tap through the menu before exiting the PDF. Maybe that should be a request for another issue ...

**Comment by Hzj-jie** [comment]
I also prefer to have a feature to export all highlights / bookmarks, and yes, even better, automatically. HouQP and I had a conversation before about the possibility. I prefer to have a background task to do so, but I am still investigating the possibility. i.e. we may not always have required tools on different platforms.

**Comment by chrox** [comment]
Preferably we can create a menu entry "save document on exit" with three options "Prompt, Save, Don't save". So that user can select one option from the GUI other than editing the global settings file.

**Comment by houqp** [comment]
Menu entry for the setting++

**Comment by houqp** [comment]
BTW, I think it's fine to just implement the three state setting in this PR. We can add the UI stuff in a separate one.

**Comment by Hzj-jie** [comment]
No problem, I will work on it soon.

**Comment by houqp** [comment]
Thanks, looks good to me, please stash and rebase 👍

**Comment by chrox** [comment]
Yes, we should never merge upstream branch locally. And `git rebase upstream/master` before each pull request is a good practice.

**Comment by houqp** [comment]
Ha, just noticed github has new feature to merge and squash from the UI with one click :)

**Comment by Hzj-jie** [comment]
Faint, I have just finished squash.

**Comment by houqp** [comment]
Next time also do --amend so you don't have to squash any more :P

**Pull request merged**